Mathlib/Data/Set/Subset.lean

```diff
@@ -0,0 +1,168 @@
+/-
+Copyright (c) 2024 Miguel Marco. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Miguel Marco
+-/
+import Mathlib.Data.Set.Function
+import Mathlib.Data.Set.Functor
+
+/-!
+# Sets in subtypes
+
+This file is about sets in `Set A` when `A` is a set.
+
+It defines notation `↓∩` for sets in a type pulled down to sets in a subtype, as an inverse
+operation to the coercion that lifts sets in a subtype up to sets in the ambient type.
+
+This module also provides lemmas for `↓∩` and this coercion.
+
+## Notation
+
+Let `α` be a `Type`, `A B : Set α` two sets in `α`, and `C : Set A` a set in the subtype `↑A`.
+
+- `A ↓∩ B` denotes `(Subtype.val ⁻¹' B : Set A)` (that is, `{x : ↑A | ↑x ∈ B}`).
+- `↑C` denotes `Subtype.val '' C` (that is, `{x : α | ∃ y ∈ C, ↑y = x}`).
+
+This notation, (together with the `↑` notation for `Set.CoeHead` in `Mathlib.Data.Set.Functor`)
+is scoped to the `Set.Notation` namespace.
+To enable it, use `open Set.Notation`.
+
+
+## Naming conventions
+
+Theorem names refer to `↓∩` as `preimage_val`.
+
+## Tags
+
+subsets
+-/
+
+open Set
+
+variable {ι : Sort*} {α : Type*} {A B C : Set α} {D E : Set A}
+variable {S : Set (Set α)} {T : Set (Set A)} {s : ι → Set α} {t : ι → Set A}
+
+namespace Set.Notation
+
+/--
+Given two sets `A` and `B`, `A ↓∩ B` denotes the intersection of `A` and `B` as a set in `Set A`.
+
+The notation is short for `((↑) ⁻¹' B : Set A)`, while giving hints to the elaborator
+that both `A` and `B` are terms of `Set α` for the same `α`.
+This set is the same as `{x : ↑A | ↑x ∈ B}`.
+-/
+scoped notation3 A:67 " ↓∩ " B:67 => (Subtype.val ⁻¹' (B : type_of% A) : Set (A : Set _))
+
+end Set.Notation
+
+namespace Set
+
+open Notation
+
+lemma preimage_val_eq_univ_of_subset (h : A ⊆ B) : A ↓∩ B = univ := by
+  rw [eq_univ_iff_forall, Subtype.forall]
+  exact h
+
+lemma preimage_val_subset_preimage_val_iff : A ↓∩ B ⊆ A ↓∩ C ↔ A ∩ B ⊆ A ∩ C := by
+  rw [inter_comm A, inter_comm A, ← Subtype.image_preimage_coe, ← Subtype.image_preimage_coe,
+    image_subset_image_iff Subtype.val_injective]
+
+lemma preimage_val_eq_iff : A ↓∩ B = A ↓∩ C ↔ A ∩ B = A ∩ C := by
+  rw [Subtype.preimage_val_eq_preimage_val_iff, inter_comm _ A, inter_comm _ A]
```

Comment:
With my suggestion above to flip the `∩` either here or in other files, this lemma becomes a duplicate of `Subtype.preimage_val_eq_preimage_val_iff`

Reply:
Now that #10433 is merged, this lemma is a duplicate.

Reply:
Fixed.